form/create_form_type_extension.rst

```diff
@@ -120,16 +89,16 @@ tag:
         use AppBundle\Form\Extension\ImageTypeExtension;
         use Symfony\Component\Form\Extension\Core\Type\FileType;
 
-        $container
-            ->register('app.image_type_extension', ImageTypeExtension::class)
+        $container->autowire(ImageTypeExtension::class)
             ->addTag('form.type_extension', array(
                 'extended_type' => FileType::class
             ))
         ;
 
-The ``extended_type`` key of the tag is the type of field that this extension should
-be applied to. In your case, as you want to extend the ``Symfony\Component\Form\Extension\Core\Type\FileType``
-field type, you will use that as the ``extended_type``.
+The ``extended_type`` key of the tag must match the class you're returning from
```

Comment:
I completely agree. But, that's the way the feature is built currently - I just double-checked it. I think that `getExtendedType()` should be static and this `extended_type` should not be needed (exactly how an event subscriber works).